cmd/kubeadm/app/cmd/util/cmdutil.go

```diff
@@ -62,18 +63,21 @@ func ValidateExactArgNumber(args []string, supportedArgs []string) error {
 	return nil
 }
 
-// FindExistingKubeConfig returns the localtion of kubeconfig
-func FindExistingKubeConfig(file string) string {
-	// The user did provide a --kubeconfig flag. Respect that and threat it as an
-	// explicit path without building a DefaultClientConfigLoadingRules object.
-	if file != kubeadmconstants.GetAdminKubeConfigPath() {
+// FindKubeConfigFile can be used to search for a kubeconfig in standard locations
+// if and empty string is passed to the function. If a non-empty string is passed
+// the function returns the same string.
+func FindKubeConfigFile(file string) string {
+	// If a value is provided respect that.
+	if file != "" {
 		return file
 	}
-	// The user did not provide a --kubeconfig flag. Find a config in the standard
-	// locations using DefaultClientConfigLoadingRules, but also consider the default config path.
+	// Find a config in the standard locations using DefaultClientConfigLoadingRules,
+	// but also consider the default config path.
 	rules := clientcmd.NewDefaultClientConfigLoadingRules()
 	rules.Precedence = append(rules.Precedence, kubeadmconstants.GetAdminKubeConfigPath())
-	return rules.GetDefaultFilename()
+	file = rules.GetDefaultFilename()
+	klog.V(1).Infof("Using kubeconfig file: %s", file)
+	return file
 }
```

Comment:
I still find It confusing that this machinery can return path to the file that doesn't exist. Should we check it here and return an error or at least warn user somehow?

Reply:
this value will eventually run a function that tries to open this file and an error will be thrown if the file is missing.

Reply:
The point is that possible error is not handled immediately. We (intentionally?) leave error handling for future processing which may or not may happen. This is generally fragile and error prone approach in my opinion.

Reply:
please show examples in the source code where error handling is delayed for kubeconfig.

Reply:
Sure. Almost any place where FindExistingConfig function is called, e.g. https://github.com/kubernetes/kubernetes/blob/master/cmd/kubeadm/app/cmd/alpha/certs.go#L171
The potential error (returned path to not existing config) is not processed in the FindExistingConfig function. It's left unprocessed in a hope that it will be processed in kubeconfigutil.ClientSetFromFile function.

Reply:
and ClientSetFromFile will fail if the file is missing.

the function calls:
https://github.com/kubernetes/kubernetes/blob/7f23a743e8c23ac6489340bbb34fa6f1d392db9d/staging/src/k8s.io/client-go/tools/auth/clientauth.go#L88
which already has a check if a file exists.

we could add another check in FindKubeConfigFile but it would be essentially a double check for all our code.

now, if this function is user facing in a way where the user should always receive an existing file path as a "string" that is printed in the terminal, then it would make sense to modify this function. except that's not what we are doing in our codebase today.

Reply:
up to you. In my opinion it would make sense to catch potential errors as soon as possible. Current code doesn't do it, leaving error processing to LoadFromFile, ClientSetFromFile, getClientset and other functions. That processing may or may not happen. I hope it happens in current Kubeadm code. However, looking at the calls of FindExistingKubeConfig in  app/cmd/upgrade/upgrade.go and a couple of similar places I would't be surprised if it doesn't happen at all.

Reply:
@fabriziopandini, @rosti What do you guys think about this? I can explain it in more details if needed.

Reply:
Depends on the use cases, which TBH, I did not check in their entirety. If we always load that file after we call this function and not do anything else with it prior to that, then we are OK. Otherwise, it will not be very safe.
If we don't need an extra check, let's not have it, but instead, we can rename the func to something, that implies getting a config file path, rather than verifying it exists. In that case words like "find" and "existing" should not be included in the function name.

Considering this, how about `s/FindKibeConfigFile/GetKubeConfigPath`?

Reply:
the rename SGTM will update.

Reply:
i just had a look and all calls to (the newly renamed) `GetKubeConfigPath` results in a kubeconfig string that is always passed to `ClientSetFromFile`.

which to me confirms this:
> If we always load that file after we call this function and not do anything else with it prior to that, then we are OK. Otherwise, it will not be very safe.